sql/core/src/main/scala/org/apache/spark/sql/execution/CacheManager.scala

```diff
@@ -185,9 +185,10 @@ class CacheManager extends Logging {
     plan match {
       case lr: LogicalRelation => lr.relation match {
         case hr: HadoopFsRelation =>
+          val prefixToInvalidate = qualifiedPath.toString
           val invalidate = hr.location.rootPaths
-            .map(_.makeQualified(fs.getUri, fs.getWorkingDirectory))
-            .contains(qualifiedPath)
+            .map(_.makeQualified(fs.getUri, fs.getWorkingDirectory).toString)
+            .exists(_.startsWith(prefixToInvalidate))
```

Comment:
why do we need the prefix resolution? I think it's useful, so that users can refresh the table path to invalidate cache for partitioned data source, but it's not related to this PR right?